docs/DEBUGGING.md

```diff
@@ -7,13 +7,142 @@
+
+```JSON
+{
+  "configurations": [
+    {
+      "type": "gdbtarget",
+      "request": "attach",
+      "name": "Eclipse CDT Remote",
+      "program": "${workspaceFolder}/build/${command:espIdf.getProjectName}.elf",
+      "initCommands": [
+        "set remote hardware-watchpoint-limit 2",
+        "mon reset halt",
+        "maintenance flush register-cache",
+        "thb app_main",
+      ],
+      "gdb": "${command:espIdf.getXtensaGdb}",
+      "target": {
+        "port": "3333"
+      },
+    }
+  ]
+}
+```
+
+where `program` and `gdb` can be resolved by extension. Some additional arguments you might use are:
+
+- `runOpenOCD`: (Default: true). Run extension openOCD Server.
+- `verifyAppBinBeforeDebug`: (Default: false) Verify that current ESP-IDF project binary is the same as binary in chip.
+- `logFile`: Absolute path to the file to log interaction with gdb.
+- `verbose`: Produce verbose log output.
+- `environment`: Environment variables to apply to the ESP-IDF Debug Adapter. It will replace global environment variables and environment variables used by the extension.
+
+```json
+"environment": {
+  "VAR": "Value"
+}
+```
+
+- `imageAndSymbols`:
+
+```json
+"imageAndSymbols": {
+  "symbolFileName": "If specified, a symbol file to load at the given (optional) offset",
+  "symbolOffset": "If symbolFileName is specified, the offset used to load",
+  "imageFileName": "If specified, an image file to load at the given (optional) offset",
+  "imageOffset": "If imageFileName is specified, the offset used to load"
+}
+```
+
+- `target`: Configuration for target to be attached.
+
+```json
+"target": {
+  "type": "The kind of target debugging to do. This is passed to -target-select (defaults to remote)",
+  "host": "Target host to connect to (defaults to 'localhost', ignored if parameters is set)",
+  "port": "Target port to connect to (defaults to value captured by serverPortRegExp, ignored if parameters is set)",
+  "parameters": "Target parameters for the type of target. Normally something like localhost:12345. (defaults to `${host}:${port}`)"
+}
+```
+
+Other arguments please review this extension's package.json `gdbtarget` debugger contribution.
+
+## Use Microsoft C/C++ Extension to Debug
+
+The user can also use [Microsoft C/C++ Extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools) to debug, the community recommend this launch.json configuration:
+
+```JSON
+{
+  "configurations": [
+    {
+      "name": "GDB",
+      "type": "cppdbg",
+      "request": "launch",
+      "MIMode": "gdb",
+      "miDebuggerPath": "${command:espIdf.getXtensaGdb}",
+      "program": "${workspaceFolder}/build/${command:espIdf.getProjectName}.elf",
+      "windows": {
+        "program": "${workspaceFolder}\\build\\${command:espIdf.getProjectName}.elf"
+      },
+      "cwd": "${workspaceFolder}",
+      "environment": [{ "name": "PATH", "value": "${config:idf.customExtraPaths}" }],
+      "setupCommands": [
```

Comment:
The suggested setup command flow does not work well. Lastly, I did something like;
```
		  "setupCommands": [
			{ "text": "set remotetimeout 10" },
		  ],
		  "postRemoteConnectCommands": [
			{ "text": "mon reset halt" },
			{ "text": "maintenance flush register-cache"},
		  ],
```